### PR TITLE
Add new global method (__webpack_init_sharing__)

### PR DESCRIPTION
**Pull request by pan-alexey**
Documentation for methods [https://webpack.js.org/concepts/module-federation/#dynamic-remote-containers](https://webpack.js.org/concepts/module-federation/#dynamic-remote-containers)
Description of types [https://github.com/webpack/webpack/blob/main/module.d.ts#L176](https://github.com/webpack/webpack/blob/main/module.d.ts#L176)

Please fill in this template.

- [*] Use a meaningful title for the pull request. Include the name of the package modified.
- [*] Test the change in your own code. (Compile and run.)
- [*] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [*] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [*] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [*] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

**Comment by pan-alexey** [comment]
PR fix this problem
Снимок экрана 2022-07-30 в 23 36 11

**Comment by pan-alexey** [comment]
Ping @use-strict, @rhonsby.

**Comment by andrewbranch** [comment]
What is webpack-env for now that webpack ships types?

**Comment by pan-alexey** [comment]
@andrewbranch thanks for questions.

When developing an application in TS, it is sometimes necessary to use module variables.
https://webpack.js.org/api/module-variables/

To use webpack method calls, special types are added. Example: https://github.com/pan-alexey/webpack-hmr-server/blob/main/package/webpack-hmr-server/tsconfig.json#L13

P/S:
Webpack does not use @types/webpack-env as a dependency.

**Comment by pan-alexey** [comment]
ping @andrewbranch, @use-strict, @rhonsby.

**Comment by pan-alexey** [comment]
ping @andrewbranch, @use-strict, @rhonsby.

File: types/webpack-env/index.d.ts [hunk 1]
```diff
@@ -309,6 +309,11 @@ declare var __webpack_hash__: any;
  */
 declare var __non_webpack_require__: any;
 
+/**
+ * Initializes the shared scope. Fills it with known provided modules from this build and all remotes
+ */
+ declare var __webpack_init_sharing__: (scope: string) => Promise<void>;
```
Comment (rbuckton) [inline types/webpack-env/index.d.ts:315]: Is `__webpack_init_sharing__` always present or is it only defined in specific cases? Would it make sense to make this optional?

```suggestion
 declare var __webpack_init_sharing__: ((scope: string) => Promise<void>) | undefined;
```

Comment (pan-alexey) [reply types/webpack-env/index.d.ts:315]: @rbuckton thanks for questions. 
__webpack_init_sharing__ will always present, just like __webpack_require__
https://github.com/webpack/webpack/blob/main/lib/APIPlugin.js#L24

**Comment by pan-alexey** [comment]
ping @rbuckton @andrewbranch, @use-strict, @rhonsby.

**Review by jakebailey — APPROVED** [review]

**Comment by andrewbranch** [comment]
@pan-alexey sorry for the delay, I was on vacation last week. My question is not about why there are module-related global types for webpack; my question is why are those types here on DefinitelyTyped when they appear to be redundant with types shipped by Webpack itself:

https://github.com/webpack/webpack/blob/9fcaa243573005d6fdece9a3f8d89a0e8b399613/module.d.ts#L189

**Comment by jakebailey** [comment]
Last I used webpack, installing webpack did not actually bring these global types in at all. I didn't know about @types/webpack-env at the time, so I ended up declaring them in my project directly.

Is this a misconfiguration on webpack's end, like an unreferenced d.ts file or declaration format? (Or, maybe they don't want that default beahvior at all, hence this other package?)

**Comment by andrewbranch** [comment]
At a glance, it looks like Webpack’s `module.d.ts` is never referenced from its `"types"` entrypoint. I assume that’s intentional, to prevent those globals from showing up in Webpack config files as soon as you import something from Webpack. The user simply has to add `"types": ["webpack/module"]` to their tsconfig.json instead (or triple-slash reference it somewhere in the program). (This explicit step would not be necessary for `@types/webpack-env` due to automatic inclusion of all `@types` when no `"types"` array is specified, but that is not reason enough to duplicate types from a package onto DT.)

**Comment by jakebailey** [comment]
Ah, I see. Curiously, `@types/webpack-env` is not mentioned at all in webpack's docs.

Given webpack's types in <=4 used to be on DT, and are now included in the package on >=5, maybe this package is a relic of those pre-upstream-types times and should be deprecated somehow? Like, make this package depend on any webpack package and just reference `webpack/module` and then we drop the package?

RE: this PR, I think this PR is fine as it's not a change in the status quo and the signature is right, but I do agree that it looks like this package shouldn't be needed anymore, besides the convenience of installing this package and the types "just working" without configuration change.

**Comment by andrewbranch** [comment]
Yes, my memory is that Webpack used to ship without any types and this package was a necessity. It now seems redundant. I agree that this PR is fine, but I wanted to understand whether it needs to continue. I do think it should be deprecated.

**Comment by pan-alexey** [comment]
I look at the package download statistics and what it depends on, it is still relevant for some packages. That is why I would like to add this global type.

**Comment by pan-alexey** [comment]
Ready to merge

**Pull request merged**